common/eventconsumption/eventvisitor.h

```diff
@@ -23,6 +23,30 @@
 // interface and the adapter to convert pulled events into visisted events.
 #include "jevent.hpp"
 
+// Abstract extension of IEventVisitor that supports decorating another visitor. Decorators may:
```

Comment:
I don't follow why this is needed, or what it provides - let's talk it through at our next 1:1